test/gtest/common/test_obj_size.cc

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2019.  ALL RIGHTS RESERVED.
+ * Copyright (C) Huawei Technologies Co., Ltd. 2020.  All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <common/test.h>
+
+extern "C" {
+#include <core/ucc_global_opts.h>
+}
+
+class test_obj_size : public ucc::test {
+};
+
+#define EXPECTED_SIZE(_obj, _size) EXPECT_EQ((size_t)_size, sizeof(_obj))
+
+UCC_TEST_F(test_obj_size, size) {
+
+#if ENABLE_DEBUG_DATA
+    UCC_TEST_SKIP_R("Debug data");
+#elif defined (ENABLE_STATS)
+    UCC_TEST_SKIP_R("Statistic enabled");
+#elif UCC_ENABLE_ASSERT
+    UCC_TEST_SKIP_R("Assert enabled");
+#else
```

Comment:
Do we really need all these checks? And will they really affect size of global config?

Reply:
This is also kind of a placeholder - I'll be using it in tests I write in the future (I hope you all will too :) ). The reason NOT to run the object size test is you often want to compare to some CACHELINE_SIZE, but only in release build (because builds with statistics or debug typically add stuff to the struct).